odoo_backup_sh/models/odoo_backup_sh.py

```diff
@@ -41,7 +41,12 @@ class BackupConfig(models.Model):
     _description = 'Backup Configurations'
     _rec_name = 'database'
 
-    DATABASE_NAMES = [(db, db) for db in odoo.service.db.list_dbs() if db != 'session_store']
+    if config['list_db']:
+        DATABASE_NAMES = [(db, db) for db in odoo.service.db.list_dbs() if db != 'session_store']
+    elif config['db_name']:
+        DATABASE_NAMES = [(db, db) for db in odoo.tools.config['db_name'].split(',') if db != 'session_store']
```

Comment:
use ``self.env.cr`` instead. ``db_name`` may be not set or has another value

Reply:
`self` is not available in this context. Also, I do not understand, which cursor from which database can be used to define field `database` field in line 53. The only database connection, that is applicable (for me at least) is https://github.com/odoo/odoo/blob/8f4b98e67fe49516c0c830524de351db67c58357/odoo/service/db.py#L374 , but then I would be better just to use rejected `force=True` solution.

Reply:
Then you need to use request.env.cr

Reply:
You need to use current database. I've explained why force=True is not an option

Reply:
I said wrong about  ``request.env.cr``. You need to make a method to compute list of available options.

 database = fields.Selection(selection=_selection_database)

Reply:
Could not find another way, where we can get databases. Checked odoo code. 
https://github.com/odoo/odoo/blob/cb2862ad2a60ff4ce66c14e7af2548fdf6fc5961/odoo/cli/server.py#L135-L152
If no db_name set, odoo works without any databases.

Also could not find code, where db_name is modified.

So, I insist in using current solution.

Reply:
request.env.cr.dbname (or db_name) is always available

Reply:
request variable (`from odoo.http import request`) is usually evaluated inside http controller. In this context, evaluating request will `RuntimeError: object unbound`

Reply:
I know. I told that few messages before: https://github.com/it-projects-llc/addons-dev/pull/688#discussion_r309234706

Reply:
https://github.com/odoo/odoo/blob/12.0/addons/mail/wizard/mail_compose_message.py#L100-L105

Reply:
Fixed using https://github.com/it-projects-llc/addons-dev/pull/688/commits/64603c1c9cf1a7f46c329a8def5a44843767c1cc